Sources/ClientRuntime/Networking/Http/CRT/CRTClientEngine.swift

```diff
@@ -217,57 +224,82 @@ public class CRTClientEngine: HTTPClient {
                                     }
                                 }
                             } catch {
-                                continuation.resume(throwing: error)
+                                logger.error(error.localizedDescription)
+                                self.safeResumeThrowContinuation(
+                                    continuationFlag: continuationFlag,
+                                    continuation: continuation,
+                                    error: error
+                                )
                             }
                         }
                     }
                 } catch {
-                    continuation.resume(throwing: error)
+                    logger.error(error.localizedDescription)
+                    self.safeResumeThrowContinuation(
+                        continuationFlag: continuationFlag,
+                        continuation: continuation,
+                        error: error
+                    )
                 }
             }
         case .version_2:
             self.logger.debug("Using HTTP/2 connection")
             let crtRequest = try request.toHttp2Request()
             return try await withCheckedThrowingContinuation { (continuation: StreamContinuation) in
-                let requestOptions = makeHttpRequestStreamOptions(request: crtRequest,
-                                                                  continuation: continuation,
-                                                                  http2ManualDataWrites: true)
-                let stream: HTTP2Stream
-                do {
-                    // swiftlint:disable:next force_cast
-                    stream = try connection.makeRequest(requestOptions: requestOptions) as! HTTP2Stream
-                    try stream.activate()
-                } catch {
-                    continuation.resume(throwing: error)
-                    return
-                }
-
-                // At this point, continuation is resumed when the initial headers are received
-                // it is now safe to write the body
-                // writing is done in a separate task to avoid blocking the continuation
-                Task { [logger] in
+                let requestOptions = makeHttpRequestStreamOptions(
+                    continuationFlag: continuationFlag,
+                    request: crtRequest,
+                    continuation: continuation,
+                    http2ManualDataWrites: true
+                )
+                Task {
```

Comment:
What is the purpose of this Task?  It contains some code that doesn't appear to be async, then creates yet another async task from inside it.

Reply:
I must've forgot to clean it up; deleted as pointed out 👍